mailbox/backup/src/main/java/org/apache/james/mailbox/backup/InternalDateExtraField.java

```diff
@@ -0,0 +1,72 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.backup;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.Date;
+import java.util.Optional;
+
+import org.apache.commons.compress.archivers.zip.ZipShort;
+
+public class InternalDateExtraField extends LongExtraField {
+
+    public static final ZipShort ID = new ZipShort(0x6F61); // "ao" in little-endian
+
+    public InternalDateExtraField() {
+        super();
+    }
+
+    public InternalDateExtraField(Optional<Date> date) {
+        super(date
+            .map(Date::toInstant)
+            .map(instant -> ZonedDateTime.ofInstant(instant, ZoneId.systemDefault()))
```

Comment:
I don't get why you need to convert this Instant again and again. Can you explain me why?
I would have expected
```
date.map(Date::getTime)
```
to be sufficient

Reply:
(instead of these 6 lines)

Reply:
Because you want a fixed time zone (utc) for something stored. What happens if I export my mails from a server in France to import it in a server in Vietnam otherwise.

Hint: you will "loose" 5 hours on all the dates if you only use "getTime".

Reply:
Ok, seems like I was wrong on this:

```
Date::getTime
Returns the number of milliseconds since January 1, 1970, 00:00:00 GMT represented by this Date object.
```

The timezone is specified hence relying completely on getTime would work.

Learned something.

Hence I do agree with @rouazana  comments.

Reply:
Yes, that's exactly why I meant, thanks.

Reply:
But we want to store UTC date time as byte array, it still need conversion from local time to UTC. DOn't you think?

Reply:
As documented, Date.getTime returns the number of milliseconds since a GMT date. So if the documentation is correct, the conversion is already done calling getTime().
(knowing that UTC ~= GMT)